src/concurrency/transaction_manager.cpp

```diff
@@ -82,6 +82,59 @@ void TransactionManager::EndTransaction(TransactionContext *current_txn) {
     current_txn->ExecOnCommitTriggers();
   }
 
+  // log RWSet and result stats
+  const auto &stats_type = static_cast<StatsType>(settings::SettingsManager::GetInt(
+      settings::SettingId::stats_mode));
+
+  if (stats_type != StatsType::INVALID) {
+    auto stats_context = stats::BackendStatsContext::GetInstance();
+
+    // update counters for each element in the RWSet
+    for (const auto &element : current_txn->GetReadWriteSet()) {
+      const auto &tile_group_id = element.first.block;
+      const auto &rw_type = element.second;
+      switch (rw_type) {
+        case RWType::READ:
+        case RWType::READ_OWN:
+          stats_context->IncrementTableReads(tile_group_id);
+          break;
+        case RWType::UPDATE:
+          stats_context->IncrementTableUpdates(tile_group_id);
+          break;
+        case RWType::INSERT:
+          stats_context->IncrementTableInserts(tile_group_id);
+          break;
+        case RWType::DELETE:
+          stats_context->IncrementTableDeletes(tile_group_id);
+          break;
+        case RWType::INS_DEL:
+          stats_context->IncrementTableInserts(tile_group_id);
+          stats_context->IncrementTableDeletes(tile_group_id);
+          break;
+        default:
+          break;
+      }
+    }
+
+    oid_t database_id = 0;
+    const auto &first_tuple = current_txn->GetReadWriteSet().cbegin();
+    if (first_tuple != current_txn->GetReadWriteSet().cend()) {
+      database_id = catalog::Manager::GetInstance().
+          GetTileGroup(first_tuple->first.block)->GetDatabaseId();
```

Comment:
Can you briefly explain whats happening here?

Reply:
The transaction result stat is indexed on database_id, but I changed the logic from what I removed in TOTM. Let me just duplicate it and add some comments to make it more clear.